### PR TITLE
Integration tests for static and pattern based mapping

### PR DESCRIPTION
**Pull request by ikbalkaya**
This PR contains integration tests for 
* Static and pattern based channel tests
* Static and pattern based message tests

### CODE REVIEW

File: src/test/java/com/ably/kafka/connect/ChannelSinkTaskIT.java
Comment (KacperKluka) [inline src/test/java/com/ably/kafka/connect/ChannelSinkTaskIT.java:80]: This is repeated in every test so I'd suggest creating another field in the class called `DEFAULT_TOPIC` or `FIRST_TOPIC` and assigning that value to it :wink:

Comment (KacperKluka) [inline src/test/java/com/ably/kafka/connect/ChannelSinkTaskIT.java:80]: This is repeated in every test so I'd suggest creating another field in the class called `DEFAULT_TOPIC` or `FIRST_TOPIC` and assigning that value to it :wink:

File: src/test/java/com/ably/kafka/connect/ChannelSinkTaskIT.java
Comment (KacperKluka) [inline src/test/java/com/ably/kafka/connect/ChannelSinkTaskIT.java:85]: This is repeated in the other tests so maybe it will be worth it extracting it to a separate method? :thinking:

File: src/test/java/com/ably/kafka/connect/ChannelSinkTaskIT.java
Comment (KacperKluka) [inline src/test/java/com/ably/kafka/connect/ChannelSinkTaskIT.java:97]: This is repeated in the other tests so maybe we could extract it to a new method? :thinking:  Something like:
```java
assertReceivedExactAmountOfMessages(1);
```

**Review by KacperKluka — COMMENTED** [review]
From the functional point of view everything is good :+1: However, a lot of things are happening in the tests which makes it hard to pinpoint what exactly is the thing we're checking. Therefore, I'd suggest to refactor the tests to make it clear which code is for the preparatory work and which one is the important operation/change that we're testing :wink:

**Review by QuintinWillison — CHANGES_REQUESTED** [review]
I've not looked at the test implementation yet as was only passing by to have a quick look at scope and time required to review this... however I have an immediate observation regarding the test class name and location.

It looks like the `IT` suffix is an existing convention in this repository, however it's a convention I've not come across in Java/Maven projects before. It's surprising to me. To the passing maintainer, I don't think it's obvious as an acronym/abbreviation.

Additionally, I'm concerned that tests identified as "integration" are located in the same package as the implementation being tested. I would only ever expect that to need to be the case if package-scoped interfaces were being tested, or the system under test is not part of the public API. For an integration test I would not expect either of these scenarios to be applicable.

I would be much less surprised were I to find this class:

- in a package named something like `com.ably.kafka.connect.integration_tests`
- named `ChannelSinkTaskTest`
- has class-level commentary making it clear what the system under test is, what the scope of testing is and what actors are involved

File: src/test/java/com/ably/kafka/connect/ChannelSinkTaskIT.java
Comment (ikbalkaya) [reply src/test/java/com/ably/kafka/connect/ChannelSinkTaskIT.java]: fedbd55a595b702fcf3d3953fe81eae1fd26040d and 10e6c1acb935b3b80738d6bcd4c65e5be0af8596

File: src/test/java/com/ably/kafka/connect/ChannelSinkTaskIT.java
Comment (ikbalkaya) [reply src/test/java/com/ably/kafka/connect/ChannelSinkTaskIT.java]: I removed this from separate tests and created another test for this https://github.com/ably/kafka-connect-ably/pull/49/commits/3f89cda22af85de3bd98766cc8f01c2b2f70a310

File: src/test/java/com/ably/kafka/connect/ChannelSinkTaskIT.java
Comment (ikbalkaya) [reply src/test/java/com/ably/kafka/connect/ChannelSinkTaskIT.java]: This is also a good idea 👍 a3fe71d06dd5f3c30903c2f87239772f864bcf3e

**Review by ikbalkaya — COMMENTED** [review]
@QuintinWillison I had added some additional integration tests to an existing test class, but I had not intended to change the current structure to keep the scope smaller. After your feedback, I have now created an `integration` package that should separate integration tests from unit tests. There are also some other empty tests that needs cleaning up. I will cover them in another PR

File: src/test/java/com/ably/kafka/connect/integration/ChannelSinkTaskTest.java
Comment (QuintinWillison) [inline src/test/java/com/ably/kafka/connect/integration/ChannelSinkTaskTest.java:2]: I recognise you've inherited this header from the code you've moved, but we've been avoiding any boilerplate-esque headers in SDK repositories up until now, relying upon the presence of `COPYRIGHT` and `LICENSE` files at repository root to do that work.

File: src/test/java/com/ably/kafka/connect/integration/ChannelSinkTaskTest.java
Comment (QuintinWillison) [inline src/test/java/com/ably/kafka/connect/integration/ChannelSinkTaskTest.java:118]: This comment (and other instances of it) is not helpful for a few reasons... it:

1. repeats what the code already clearly tells us - it's superfluous - also, if `TIMEOUT` is change in the future then all instances of this comment will be incorrect
2. suggests to the reader that it "waits for 5 seconds", when that's technically wrong - it does wait but returns once the requisite number of messages have been received, therefore the time element is a _timeout_ 

I would remove these comments.

**Review by QuintinWillison — APPROVED** [review]
Happy to approve as more tests is only a good thing.

I'll leave it up to you, @ikbalkaya, as to whether you wish to act upon my minor comments...

File: src/test/java/com/ably/kafka/connect/integration/ChannelSinkTaskTest.java
Comment (ikbalkaya) [reply src/test/java/com/ably/kafka/connect/integration/ChannelSinkTaskTest.java]: That's a good idea. I removed Copyright wording from header here https://github.com/ably/kafka-connect-ably/pull/49/commits/aaebb05cade80b55b6d4fefe0ed8aa4b8fcaabf5
I also updated copyright information ab34e5c6a5fcff450d5fe41f408072a28c73a090

File: src/test/java/com/ably/kafka/connect/integration/ChannelSinkTaskTest.java
Comment (ikbalkaya) [reply src/test/java/com/ably/kafka/connect/integration/ChannelSinkTaskTest.java]: This is also a good point 👍 I removed comments here https://github.com/ably/kafka-connect-ably/pull/49/commits/7d8905a51dd79bcd8e70820cfa05d020c7a029d1

**Pull request merged**
